config/config.go

```diff
@@ -97,6 +106,23 @@ type ReplicationOptionsProtection struct {
 	Incremental string `yaml:"incremental,optional,default=guarantee_resumability"`
 }
 
+func (l *RecvOptions) SetDefault() {
+	*l = RecvOptions{Properties: &PropertyRecvOptions{}}
+}
+
+type PropertyRecvOptions struct {
+	Inherit  []zfsprop.Property          `yaml:"inherit,optional"`
+	Override map[zfsprop.Property]string `yaml:"override,optional"`
+}
+
+var _ yaml.Defaulter = (*PropertyRecvOptions)(nil)
+
+func (l *PropertyRecvOptions) SetDefault() {
+	//*l = PropertyRecvOptions{}
+	//TODO: is below necessary?
```

Comment:
i think it is, and you should have test cases where `inherit` is set, `override` isn't, and check that the code that consumes PropertyRecvOptions can deal with `Override` being nil.